pkg/database/migrations/v_0010/model.go

```diff
@@ -0,0 +1,315 @@
+	UserID         string         `gorm:"<-:create;type:varchar(256)"`
+	Status         Status         `gorm:"type:varchar(9);check:status IN ('SCHEDULED', 'FAILED', 'FINISHED', 'RUNNING', 'KILLED')"`
+	StartTime      sql.NullInt64  `gorm:"<-:create;type:bigint"`
+	EndTime        sql.NullInt64  `gorm:"type:bigint"`
+	SourceVersion  string         `gorm:"<-:create;type:varchar(50)"`
+	LifecycleStage LifecycleStage `gorm:"type:varchar(20);check:lifecycle_stage IN ('active', 'deleted')"`
+	ArtifactURI    string         `gorm:"<-:create;type:varchar(200)"`
+	ExperimentID   int32
+	Experiment     Experiment
+	DeletedTime    sql.NullInt64  `gorm:"type:bigint"`
+	RowNum         RowNum         `gorm:"<-:create;index"`
+	Params         []Param        `gorm:"constraint:OnDelete:CASCADE"`
+	Tags           []Tag          `gorm:"constraint:OnDelete:CASCADE"`
+	Metrics        []Metric       `gorm:"constraint:OnDelete:CASCADE"`
+	LatestMetrics  []LatestMetric `gorm:"constraint:OnDelete:CASCADE"`
+}
+
+type RowNum int64
+
+func (rn *RowNum) Scan(v interface{}) error {
+	nullInt := sql.NullInt64{}
+	if err := nullInt.Scan(v); err != nil {
+		return err
+	}
+	*rn = RowNum(nullInt.Int64)
+	return nil
+}
+
+func (rn RowNum) GormDataType() string {
+	return "bigint"
+}
+
+func (rn RowNum) GormValue(ctx context.Context, db *gorm.DB) clause.Expr {
+	if rn == 0 {
+		return clause.Expr{
+			SQL: "(SELECT COALESCE(MAX(row_num), -1) FROM runs) + 1",
+		}
+	}
+	return clause.Expr{
+		SQL:  "?",
+		Vars: []interface{}{int64(rn)},
+	}
+}
+
+type Param struct {
+	Key   string `gorm:"type:varchar(250);not null;primaryKey"`
+	Value string `gorm:"type:varchar(500);not null"`
+	RunID string `gorm:"column:run_uuid;not null;primaryKey;index"`
+}
+
+type Tag struct {
+	Key   string `gorm:"type:varchar(250);not null;primaryKey"`
+	Value string `gorm:"type:varchar(5000)"`
+	RunID string `gorm:"column:run_uuid;not null;primaryKey;index"`
+}
+
+type Metric struct {
+	Key       string  `gorm:"type:varchar(250);not null;primaryKey"`
+	Value     float64 `gorm:"type:double precision;not null;primaryKey"`
+	Timestamp int64   `gorm:"not null;primaryKey"`
+	RunID     string  `gorm:"column:run_uuid;not null;primaryKey;index"`
+	Step      int64   `gorm:"default:0;not null;primaryKey"`
+	IsNan     bool    `gorm:"default:false;not null;primaryKey"`
+	Iter      int64   `gorm:"index"`
+	ContextID uint    `gorm:"not null;primaryKey"`
+	Context   Context
+}
+
+type LatestMetric struct {
+	Key       string  `gorm:"type:varchar(250);not null;primaryKey"`
+	Value     float64 `gorm:"type:double precision;not null"`
+	Timestamp int64
+	Step      int64  `gorm:"not null"`
+	IsNan     bool   `gorm:"not null"`
+	RunID     string `gorm:"column:run_uuid;not null;primaryKey;index"`
+	LastIter  int64
+	ContextID uint `gorm:"not null;primaryKey"`
+	Context   Context
+}
+
+// JSONB defined JSONB data type, need to implements driver.Valuer, sql.Scanner interface
+type JSONB json.RawMessage
```

Comment:
can we move all the JSONB declarations to some other location that's not duplicated so much? Eg, `migrations/types.go` or the like?